modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java

```diff
@@ -133,7 +154,7 @@ public Map<String, Object> getDDMFormContext(
 		}
 
 		return _ddmFormRenderer.getDDMFormTemplateContext(
-			ddmForm, structureVersion.getDDMFormLayout(),
+			ddmForm, latestApprovedStructureVersion.getDDMFormLayout(),
```

Comment:
@lazhcsilva Unfortunately we can't do that. If we do, the behavior when deleting a field will change to not show deleted fields when editing an old entry:
[new_behavior.webm](https://user-images.githubusercontent.com/34186632/196712527-6cb9e101-7d1a-43bf-b3ed-64e1f6d7cbb1.webm)

But the expected behavior is to show the removed field (with the label `removed`) when editing an old entry:
[expected_behavior.webm](https://user-images.githubusercontent.com/34186632/196712554-622c53fd-4188-47c3-aff8-ca8873ed01e6.webm)


Could you please fix this? Feel free to reach out to the team if you need any help with this.